data/src/main/java/com/linkedin/data/schema/validation/ValidateDataAgainstSchema.java

```diff
@@ -142,6 +144,53 @@ protected void validate(DataElement element)
       {
         validateIterative(element);
       }
+
+      if (_toTrim.size() > 0) {
+        for (ToTrim toTrim: _toTrim) {
+          toTrim.trim();
+        }
+      }
+    }
+
+    /**
+     * Holds a data elements for later removal.
+     *
+     * This is a specialized version of {@link com.linkedin.data.it.Remover.ToRemove} which is and
+     * should remain private, so could not be used here.
+     */
+    private class ToTrim
+    {
+      private ToTrim(DataElement element)
+      {
+        DataElement parentElement = element.getParent();
+        _path = element.path();
```

Comment:
Just like done in `ToRemove`, we cannot hold a reference to DataElement, so we instead capture the fields we need from it.